shremote.py

```diff
@@ -524,60 +547,44 @@ def __init__(self, program_name, cmd_cfg, index = None):
 
         if 'begin' in cmd_cfg:
             begin_raw = cmd_cfg.formatted('begin')
-            try:
-                begin_evaled = safe_eval(begin_raw, 'begin')
-                if str(begin_evaled) != str(begin_raw): # Means that 'eval' did something
-                    log_warn("Specifying evaluatable begin without $(...) is deprecated: {}".format(begin_raw))
-                self.begin = float(begin_evaled)
-            except TypeError: # Begin shouldn't have to be eval'd again
-                self.begin = float(begin_raw)
+            self.begin = int(try_eval(begin_raw, 'begin'))
 
         self.index = index if index is not None else 0
         self.program = Program.get(program_name)
-
         self.log = Log.get(program_name)
         self.name = program_name
 
+        if self.program.start is None:
+            log_fatal("{}: Program requires a 'start' command".format(program_name))
+
         if 'duration' in cmd_cfg:
+            if self.program.stop is None:
+                log_fatal("{}: Must specify a 'stop' command if duration is specified".format(
```

Comment:
:( 
There are some times where duration is used as a flag to the program, and there is no explicit way that the program should be stopped.
This is the case with the rate client, for example. You provide a duration to the program, and the program stops after a certain amount of time. It does not need a `stop` command

Reply:
Rather, than just bound to the "duration" behavior, I think a stop command should be mandatory in all circumstances, such that it can be used when  shremote is stopped in the middle of execution with crtl+C

Reply:
Hm, that's interesting, and not how I've been using the stop command until this point.
Do you think the stop command should always run after the duration, even if the program has stopped on its own?
I've had a couple of cases where there are multiple instances of, e.g., rate client running. I need the attack traffic to stop, but normal traffic to continue. In that case I depend on the rate client to end normally, because a `pkill` would kill both instances of the client.
I'm not sure how to deal with that scenario if `stop` is always run.

Reply:
Stop could run in the following circumstances:
- The user sends SIGINT to Shremote
- Enforce_duration is requested, and the program has been running for longer than the specified duration

This supports the behavior you just described.

Btw I think there is a (understandable) mix about two concepts of "duration": one is the  argument given to a program, an the other the information shremote uses to oversee a program. We might want to work on making the two independent.

Reply:
There might be a better solution involving killing the process with popen that will get rid of the need for providing an explicit kill command for most processes. I might want to keep that out of this PR though

Reply:
So for this one do we want to keep a stop command required in the presence of enforce_duration? I think that was the previous behavior.

Also there is another case where we'd already want a stop command all the time, that is when log_fatal is invoked, for any reason, and attempts to kill everything

Reply:
I agree that would be a good feature, but at the moment it would break a fair number of my configs, so I'd rather wait and not change this behavior, if its lack does not affect your use of the tool